lib/builders/gpu-builder.ts

```diff
@@ -0,0 +1,133 @@
+import * as blueprints from '../../lib';
+import * as utils from "../utils";
+import { NestedStack, NestedStackProps } from 'aws-cdk-lib';
+import { Construct } from 'constructs';
+import * as eks from "aws-cdk-lib/aws-eks";
+import * as ec2 from "aws-cdk-lib/aws-ec2";
+import { NodegroupAmiType } from 'aws-cdk-lib/aws-eks';
+
+/**
+ * Configuration options for GPU Builder.
+ */
+export interface GpuOptions {
+    /**
+     * Required, Kubernetes version to use for the cluster.
+     */
+    kubernetesVersion: eks.KubernetesVersion,
+    /** 
+     * Required, Instance class to use for the cluster. 
+     */
+    instanceClass: ec2.InstanceClass,
+    /** 
+     * Required, Instance size to use for the cluster. 
+     */
+    instanceSize: ec2.InstanceSize,
+    /** 
+     * Optional, AMI Type for GPU Nodes @ default AL2_X86_64_GPU 
+     */
+    gpuAmiType?: NodegroupAmiType,
+    /** 
+     * Optional, Desired number of nodes to use for the cluster. 
+     */
+    desiredNodeSize?: number,
+    /** 
+     * Optional, Minimum number of nodes to use for the cluster. 
+     */
+    minNodeSize?: number,
+    /** 
+     * Optional, Maximum number of nodes to use for the cluster. 
+     */
+    maxNodeSize?: number,
+    /** 
+     * Optional, Block device size.
+     */
+    blockDeviceSize?: number,
+    /**
+     * Optional, Cluster Provider Tags.
+     */
+    clusterProviderTags?: {
+        [key: string]: string;
+    },
+
+    /**
+     * Optional, Node Group Tags for AL2 nodes 
+     * which run standard cluster software.
+     */
+    nodeGroupTags?: {
+        [key: string]: string;
+    }
+}
+
+/**
+ * Default props to be used when creating the GPU nodes 
+ * for EKS cluster
+ */
+const defaultOptions: GpuOptions = {
```

Comment:
there is a warning the defaultOptions are assigned value but never used. 
1. make sure `make lint` is success
2. Why do we have all the default options specified but never use them? is it an indication of a defect?

Reply:
Fixed